src/win/wwindow.c

```diff
@@ -959,31 +974,43 @@ static LRESULT CALLBACK window_callback(HWND hWnd, UINT message,
              * We have to create the resize event, so the application could
              * redraw its content.
              */
-            if (!resize_postponed) {
-               resize_postponed = true;
-               _beginthread(postpone_thread_proc, 0, (void *)d);
-            }
+            generate = true;
+         }
+         al_lock_mutex(resize_event_thread_mutex);
+         if (_al_vector_contains(&resizing_displays, &win_display)) {
+            generate = false;
+         }
+         al_unlock_mutex(resize_event_thread_mutex);
+         if (generate) {
+            win_generate_resize_event(win_display);
          }
          return 0;
-      case WM_ENTERSIZEMOVE:
+      }
+      case WM_ENTERSIZEMOVE: {
          /* DefWindowProc for WM_ENTERSIZEMOVE enters a modal loop, which also
           * ends up blocking the loop in d3d_display_thread_proc (which is
           * where we are called from, if using D3D).  Rather than batching up
           * intermediate resize events which the user cannot acknowledge in the
           * meantime anyway, make it so only a single resize event is generated
           * at WM_EXITSIZEMOVE.
           */
+         al_lock_mutex(resize_event_thread_mutex);
          if (d->flags & ALLEGRO_DIRECT3D_INTERNAL) {
-            resize_postponed = true;
+            win_display->ignore_resize = true;
```

Comment:
Use a different variable, so that al_resize_display can be ostensibly called while live resize is happening and we don't get out of sync.